src/ckeditor/ckeditor.component.spec.ts

```diff
@@ -151,6 +151,44 @@ describe( 'CKEditorComponent', () => {
 						expect( component.instance.config.width ).toBe( 1000 );
 						expect( component.instance.config.height ).toBe( 1000 );
 					} );
+
+					it( 'editor should have undo plugin', () => {
+						expect( component.instance.plugins.undo ).not.toBeUndefined();
+					} );
+
+					it( 'should register changes', () => {
+						const spy = jasmine.createSpy();
+
+						component.registerOnChange( spy );
+						component.instance.setData( '<p>Hello World!</p>' );
+						component.instance.setData( '</p>I am CKEditor for Angular!</p>' );
+
+						expect( spy ).toHaveBeenCalledTimes( 2 );
+					} );
+				} );
+
+				describe( 'when set without undo plugin', () => {
+					beforeEach( ( done ) => {
+						component.config = {
+							removePlugins: 'undo'
+						};
+						fixture.detectChanges();
+						whenEvent( 'ready', component ).then( done );
+					} );
+
+					it( 'editor should not have undo plugin', () => {
+						expect( component.instance.plugins.undo ).toBeUndefined();
+					} );
+
+					it( 'should register changes without undo plugin', () => {
+						const spy = jasmine.createSpy();
+
+						component.registerOnChange( spy );
+						component.instance.setData( '<p>Hello World!</p>' );
+						component.instance.setData( '</p>I am CKEditor for Angular!</p>' );
+
+						expect( spy ).toHaveBeenCalledTimes( 2 );
+					} );
```

Comment:
Could you also add test which check if `ngModelChange` event is fired (for both undo and no-undo plugin scenarios)?

This is the general event which is fired upon editor content changes (only if `ngModel` is set) and we should make sure it is fired in both cases (spoiler: checked on demo and it works, but still we should have test coverage for it).
The event is fired by `this.onChange( ... )` call in our `propagateChange` method (https://github.com/ckeditor/ckeditor4-angular/pull/56/files#diff-756e5674b9c69ad85fad09a2b0c3e852R306) and you need `ngModel` property ( e.g. `[(ngModel)]="mycontent"`) set to activate it (similarly also as in initial bug report sample https://stackblitz.com/edit/ckeditor4-orm88s).

Reply:
Actually I found it quite challenging to write test explicitly checking if `ngModelChange` was fired 🤔 So for now I decided to test `onChange` event. Another way that seems possible is to add `(ngModelChange)` attribute to editor component with a custom function and then we should be able to spy on that function calls. Or maybe you can come up with something better 👐

Reply:
From what I understand `onChange` triggers `ngModelChange` so we should be safe with such test :+1: